pal03377/Athena-New
# Add LLaMA 2 through Replicate and improve config for text module

### Motivation and Context



We want to evaluate on LLaMA 2 models and there were some errors with the environment when setting up the LLM api keys.

### Description

- Fix some issues with the environment in the OpenAI config
- Add Replicate as provider with the three LLaMA 2 chat models I want to evaluate on

### Issues
- [ ] When switching between providers, like OpenAI -> Replicate -> OpenAI some default parameters are unset
     - I don't know how to fix it, maybe just skip it. It is just a bit of a worse UX
- [x] Prospector union type error

### Steps for Testing

1. Setup some placeholder api key i.e. "XXXXXX" for replicate (or create an account and link credit card on https://replicate.com)
2. Check out the config schema in the playground
3. Optional: Module requests generate feedback suggestions with Replicate

### Screenshots


<img width="1285" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/5f86427e-c17a-40cc-8b9c-501fad9c828a">
